rocketchat-core/src/main/java/com/rocketchat/core/RestImpl.java

```diff
@@ -37,12 +44,60 @@
     private final HttpUrl baseUrl;
     private final TokenProvider tokenProvider;
     private final Moshi moshi;
+    private final Logger logger;
 
     RestImpl(OkHttpClient client, Moshi moshi, HttpUrl baseUrl, TokenProvider tokenProvider, Logger logger) {
         this.client = client;
         this.moshi = moshi;
         this.baseUrl = baseUrl;
         this.tokenProvider = tokenProvider;
+        this.logger = logger;
+    }
+
+    void signin(String username, String password, final LoginCallback loginCallback) {
```

Comment:
Next time, if you are going to reorder methods (which is a good thing, don't get me wrong), do in a separate PR just for that... New code mixed with code reordering makes it hard to review :smiley:

Reply:
Ohh, sorry! I'll pay attention on it next time!